File: CodeChef/Ranking 1000_1200/SplitN/mySolution.py
t=int(input())
while t > 0 :
    t -= 1
    x=int(input())
    n=0
    while x > 0 :
        a=0
        c=0
        while a <= x:
            a = 2**c
            c += 1
        a /= 2
        x -= a
        if x > 0 :
            n += 1
    print(n)

File: CodeChef/Ranking 1000_1200/ATMMachine/mySolution.java
import java.util.Scanner;
class Codechef
{
	public static void main (String[] args)
	{
	    
        Scanner input = new Scanner(System.in);
	    int T;
	    T = input.nextInt();
	    int N,K,A;
	    for(T=T; T>0; T--)
	    {
	        N=input.nextInt();
	        K=input.nextInt();
	        for(N=N; N>0; N--)
	        {
	            A=input.nextInt();
	            if(A<=K)
	            {
	                System.out.print("1");
	                K-=A;
	            }else{
	                System.out.print("0");
	            }
	        }
	        System.out.print("\n");
	    }
	}
}
